### ISSUE TITLE
False positive: Unused function in Vue SFC script section used in template

### ISSUE THREAD

**Issue by asfaarahmed** [biomejs/biome#7139]
### Environment information

```block
OS: macOS / Windows / Linux
Node: v24.4.1
NPM: 11.4.2
Biome version: 2.1.1 (or latest)
```

### What happened?

Biome incorrectly reports a function as unused in a Vue SFC even though it's used in the template section.

```
<template>
  <button @click="handleClick">Click Me</button>
</template>

<script setup lang="ts">
function handleClick() {
  console.log('Clicked!');
}
</script>
```

**biome.jsonc**
```
{
  "$schema": "https://biomejs.dev/schemas/2.1.1/schema.json",
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true
  },
  "files": {
    "ignoreUnknown": false
  },
  "formatter": {
    "enabled": true,
    "useEditorconfig": true,
    "formatWithErrors": false,
    "indentStyle": "space",
    "indentWidth": 2,
    "lineEnding": "lf",
    "lineWidth": 120,
    "attributePosition": "auto",
    "bracketSpacing": true
  },
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "complexity": {
        // Opposing `Array.prototype.forEach` in favor of for..of is a bit much
        // for a linting rule. There are cases where `forEach` are perfectly
        // legitimate.
        // See: https://biomejs.dev/linter/rules/no-for-each/
        "noForEach": "off"
      },
      "correctness": {
        // Undeclared variables are not reported as a problem by default, but
        // this is a behavior we want to enforce. Note that build flags need to
        // be listed as available globals in the `javascript.global` array.
        // See: https://biomejs.dev/linter/rules/no-undeclared-variables/
        // "noUndeclaredVariables": "error",
        // Unused imports are not reported as a problem by default, but this
        // is a behavior we want to enforce.
        // See: https://biomejs.dev/linter/rules/no-unused-imports/
        // "noUnusedImports": "error",
        // Unused variables are not reported as a problem by default, but this
        // is a behavior we want to enforce. We have a SCM to keep track of
        // unused code and should not resort to leaving it in the code base.
        // See: https://biomejs.dev/linter/rules/no-unused-variables/
        // "noUnusedVariables": "error"
      },
      "performance": {
        // Using the spread operator on `Array.prototype.reduce` accumulators
        // is conceptually much slower than mutating the accumulator variable
        // because it creates new data with every pass on the array. That being
        // said, it’s not our performance bottleneck and we have a lot of cases
        // where we do that, so we turn it off.
        // See: https://biomejs.dev/linter/rules/no-accumulating-spread/
        "noAccumulatingSpread": "off" // fix/get rid of this
      },
      "style": {
        "noParameterAssign": "off" // fix/get rid of this
      },
      "suspicious": {
        "noImplicitAnyLet": "off" // fix/get rid of this
      }
    }
  },
  "javascript": {
    "formatter": {
      "jsxQuoteStyle": "single",
      "quoteProperties": "asNeeded",
      "trailingCommas": "all",
      "semicolons": "always",
      "arrowParentheses": "asNeeded",
      "bracketSameLine": false,
      "quoteStyle": "single",
      "attributePosition": "auto",
      "bracketSpacing": true
    }
  }
}
```

### Expected result

Biome should detect that handleClick is used in the template and not report it as unused.

**Comment by dyc3** [comment]
This is expected behavior because our support for Vue is only partial. We only parse the script section right now. We recommend disabling this rule and a couple others for .vue files

See: https://biomejs.dev/internals/language-support/#html-super-languages-support
